ecommerce/src/components/AlgoliaSearch/AlgoliaSearch.tsx

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import {
+  InstantSearch,
+  SearchBox,
+  RefinementList,
+  connectHits,
+} from "react-instantsearch-dom";
+import { client } from "../../../config/algolia";
+
+interface HitData {
+  previewUrl: string | undefined;
+  objectID: string;
+  data: {
+    images?: { image: string }[];
+    productName: string;
+    category: string;
+    price: number;
+    description: string;
+    sizes?: { label: string }[];
+  };
+}
+
+const EditHits = ({ hits }: { hits: HitData[] }) => {
+  return (
+    <div className="hits-container flex flex-wrap gap-4">
+      {hits.slice(0, 4).map((hit: any) => (
+        <Hit key={hit.objectID} hit={hit} />
+      ))}
+    </div>
+  );
+};
+
+const ConnectedHits = connectHits(EditHits);
+
+function SearchComponent() {
+  return (
+    <div className="flex flex-col items-center justify-center p-4">
+      <InstantSearch searchClient={client} indexName="builder-product-data">
+        <div className="w-full max-w-xl">
+          <div className="flex justify-center">
+            <SearchBox
+              translations={{
+                placeholder: "Search for products",
+              }}
+            />
+          </div>
+          <p className="text-center text-md mt-3">
+            Search by name, category, color, or size
+          </p>
+        </div>
+
+        <div className="filters mb-8 w-full max-w-xl flex flex-col items-center">
+          <RefinementList attribute="category" className="w-full" />
+          <RefinementList attribute="data.colors.label" className="w-full" />
+          <RefinementList attribute="data.sizes.label" className="w-full" />
+        </div>
+
+        <ConnectedHits />
+      </InstantSearch>
+    </div>
+  );
+}
+
+function Hit({ hit }: { hit: HitData }) {
+  if (!hit || Object.keys(hit).length === 0) {
+    return null;
+  }
+  return (
+    <div className="hit-container w-64 border p-4 mb-4 flex flex-col items-center">
```

Comment:
@VPilipenko334  also can we use the ProductCard component here to match the other styles?  the algolia data is based on our product data model I believe so should be able to just pop in the product